tee-worker/ts-tests/examples/direct-invocation/util.ts

```diff
@@ -0,0 +1,194 @@
+import { createPublicKey, KeyObject } from 'crypto';
+import WebSocketAsPromised from 'websocket-as-promised';
+
+export function toBalance(amountInt: number) {
+    return new BN(amountInt).mul(new BN(10).pow(new BN(12)));
+}
+
+// Send the request to worker ws
+// we should perform different actions based on the returned status:
+//
+// `Submitted`:
+// the request is submitted to the top pool, we should start to subscribe to parachain headers to wait for async parachain event
+//
+// `InSidechainBlock`
+// the request is included in a sidechain block: the state mutation of sidechain is done, the promise is resolved
+// the corresponding parachain event should be emitted **around** that, it's not guaranteed if it's before or after this status
+// due to block inclusion delays from the parachain
+//
+async function sendRequest(
+    wsClient: WebSocketAsPromised,
+    request: any,
+    api: ApiPromise
+): Promise<WorkerRpcReturnValue> {
+    const p = new Promise<WorkerRpcReturnValue>((resolve) =>
+        wsClient.onMessage.addListener((data) => {
+            let result = JSON.parse(data.toString()).result;
+            const res = api.createType('WorkerRpcReturnValue', result).toJSON() as WorkerRpcReturnValue;
+            console.log('response status: ', res.status);
+            if (res.status === 'Error') {
+                throw new Error('ws response error: ' + res.value);
+            }
+            // resolve it once `do_watch` is false, meaning it's the final response
+            if (!res.do_watch) {
+                // TODO: maybe only remove this listener
+                wsClient.onMessage.removeAllListeners();
+                resolve(res);
+            } else {
+                // TODO: subscribe to parachain headers if the status is `Submitted`
+            }
+        })
+    );
+
+    wsClient.sendRequest(request);
+    return p;
+}
+
+// TrustedCalls are defined in:
+// https://github.com/litentry/litentry-parachain/blob/d4be11716fdb46021194bbe9fe791b15249a369e/tee-worker/app-libs/stf/src/trusted_call.rs#L61
+//
+// About the signature, it's signed with `KeyringPair` here.
+// In reality we need to get the user's signature on the `payload`.
+export const createSignedTrustedCall = (
+    parachain_api: ApiPromise,
+    trustedCall: [string, string],
+    account: KeyringPair,
+    // hex-encoded mrenclave, retrieveable from parachain enclave registry
+    // TODO: do we have a RPC getter from the enclave?
+    mrenclave: string,
+    nonce: Codec,
+    params: Array<any>
+) => {
+    const [variant, argType] = trustedCall;
+    const call = parachain_api.createType('TrustedCall', {
+        [variant]: parachain_api.createType(argType, params),
+    });
+    const payload = Uint8Array.from([
+        ...call.toU8a(),
+        ...nonce.toU8a(),
+        ...hexToU8a(mrenclave),
+        ...hexToU8a(mrenclave), // should be shard, but it's the same as MRENCLAVE in our case
```

Comment:
"in our case" here means our Parachain regardless of the network, right? How likely it is to change in the future?

Reply:
Yes it's network unrelated.
> How likely it is to change in the future?

Highly improbable, even if it's going to change, that's not in near future